src/session-feedback/session-feedback.service.ts

```diff
@@ -0,0 +1,51 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { PartnerAccessEntity } from 'src/entities/partner-access.entity';
+import { PartnerEntity } from 'src/entities/partner.entity';
+import { SessionFeedbackEntity } from 'src/entities/session-feedback.entity';
+import { SessionEntity } from 'src/entities/session.entity';
+import { SubscriptionUserEntity } from 'src/entities/subscription-user.entity';
+import { SubscriptionEntity } from 'src/entities/subscription.entity';
+import { TherapySessionEntity } from 'src/entities/therapy-session.entity';
+import { UserEntity } from 'src/entities/user.entity';
+import { Repository } from 'typeorm';
+import { SessionService } from '../session/session.service';
+import { SessionFeedbackDto } from './dtos/session-feedback.dto';
+
+@Injectable()
+export class SessionFeedbackService {
+  constructor(
+    @InjectRepository(PartnerAccessEntity)
```

Comment:
I am not exactly sure why all these repos are needed, as it does not seem to be the case for other services in this repo, but I had to add all of these to get rid of any errors created. The only other service that this service directly depends on is the session service - the rest of the dependencies are just transitive

Reply:
It's a great question and you are right to pick it up. It is something we are slowly trying to fix. The issue is the function getUserByFirebaseId in the userService. It does waayyyyyy too much. It's unfortunately used in the AuthGuard. The FirebaseAuthGuard is used across all the services. What we are trying to do is start to divide up how we fetch the data and ensure everything is fetched separately. See this [issue](https://github.com/chaynHQ/bloom-backend/issues/430). It's going to be a slow process.